omen2/table.py

```diff
@@ -282,7 +282,7 @@ def __clean_cache(self, where, db_pks):
                 remove_from_cache.add(k)
         for pop_me in remove_from_cache:
             log.debug("removing %s from cache", pop_me)
-            self._cache.pop(pop_me)
+            self._cache.pop(pop_me, None)  # Don't raise if not in cache
```

Comment:
Seems like a reasonable fix for a race.

Possible to repro in a test?

Reply:
I tried, but I couldn't.